cmd/kubeadm/app/phases/controlplane/volumes.go

```diff
@@ -70,6 +70,13 @@ func getHostPathVolumesForTheControlPlane(cfg *kubeadmapi.MasterConfiguration) c
 	// Read-only mount for the controller manager kubeconfig file
 	controllerManagerKubeConfigFile := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.ControllerManagerKubeConfigFileName)
 	mounts.NewHostPathMount(kubeadmconstants.KubeControllerManager, kubeadmconstants.KubeConfigVolumeName, controllerManagerKubeConfigFile, controllerManagerKubeConfigFile, true, &hostPathFileOrCreate)
+	// Read-only mount of the cloud config file if present
+	if cfg.CloudProvider != "" {
+		if _, err := os.Stat(DefaultCloudConfigPath); err == nil {
+			mounts.NewHostPathMount(kubeadmconstants.KubeAPIServer, kubeadmconstants.KubeCloudConfigVolumeName, DefaultCloudConfigPath, DefaultCloudConfigPath, true, &hostPathFileOrCreate)
```

Comment:
Should `hostPathFileOrCreate` be `hostPathDirectoryOrCreate`?
or, if we use `hostPathFileOrCreate`, we should mount the cloud config file, not the directory.

Reply:
After taking another looking at the code, seems like `DefaultCloudConfigPath = "/etc/kubernetes/cloud-config"` is a file? Hmm, the file name is a bit confusing...

Reply:
should be FileOrCreate yes

Reply:
Is this REALLY needed for the API server? Is there actually anything in the API server that can consume cloud code anymore...? I don't think so